controllers/common/common.go

```diff
@@ -201,3 +204,25 @@ func GetNumWorkers(listLength int, concurrencyPerPolicy int) int {
 
 	return numWorkers
 }
+
+// TypeConverter is a helper function to converter type struct a to b
+func TypeConverter(a, b interface{}) error {
+	js, err := json.Marshal(a)
+	if err != nil {
+		return err
+	}
+
+	return json.Unmarshal(js, b)
+}
+
+var (
+	matchFirstCap = regexp.MustCompile("(.)([A-Z][a-z]+)")
+	matchAllCap   = regexp.MustCompile("([a-z0-9])([A-Z])")
+)
+
+func ToSnakeCase(str string) string {
+	snake := matchFirstCap.ReplaceAllString(str, "${1}_${2}")
```

Comment:
Could you please explain in this comment how this works? I'm not quite following.